org.karapulse.Karapulse.json

```diff
@@ -0,0 +1,114 @@
+{
+    "app-id": "org.karapulse.Karapulse",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.32",
+    "sdk": "org.gnome.Sdk",
+    "command": "karapulse",
+    "cleanup": [
+        "/sdk",
+        "/lib/node_modules",
+        "/bin/npm",
+        "/bin/node",
+        "/bin/npx"
+    ],
+    "finish-args": [
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--filesystem=host",
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=wayland",
+        "--socket=pulseaudio",
+        "--device=dri"
+    ],
+    "build-options": {
+        "append-path": "/app/sdk/rust/bin",
+        "build-args": [
```

Comment:
We don't allow network during build here in Flathub. You should use ninja dist to create a tarball and use it on the stable manifest.

Reply:
But won't `dist` need the network then?

So this is different that what your template is doing then, right?

Reply:
the dist will create a tarball that you can include in your own stable release. My template lacks documentation regarding that, I wanted to update that, but I always forget.

Reply:
I opened https://gitlab.gnome.org/bilelmoussaoui/gtk-rust-template/issues/1 so we'll keep track of updating the template then.

Reply:
I don't know what ninja dist is, but there is also cargo-vendor and a generator in [flatpak-builder-tools](https://github.com/flatpak/flatpak-builder-tools/blob/master/cargo/flatpak-cargo-generator.py).

Reply:
Ninja dist calls cargo vendor in the template you're using. You will need to bundle the dependencies required by the server..we have a tool for that in flatpak builder tools

Reply:
Ok so if I understand properly I have two options:
- Generate a tarball with all the deps (cargo, npm) and point my manifest to it rather than git
- Use `flatpak-cargo-generator.py` and `flatpak-npm-generator.py` to include all the deps in the manifest

Is that correct? What would be the most idiomatic way?

Reply:
Ninja dist is the way to go for cargo as cargo generator doesn't seem to work that well, you will need to include the npm dependencies with npm generator as ninja dist doesn't do that unless you change the dist script in build-aux